puppet/modules/quickstack/manifests/pacemaker/ceilometer.pp

```diff
@@ -52,6 +52,18 @@
       Anchor['redis-begin']
       Exec['redis-master-is-up'] ->
       Exec['i-am-ceilometer-vip-OR-ceilometer-is-up-on-vip']
+
+      quickstack::pacemaker::constraint::base {"redis-then-ceilometer-central-contraint":
```

Comment:
True, constraints.pp was meant to be more inter-service but I think it is OK here.  That's because redis only gets setup if ceilometer does (assuming $coordination_backend == 'redis' which you would then also have to check for in constraints.pp if you moved it the declaration there).  If there was an include_redis flag in params.pp, I'd be more inclined to have it in constraints.pp.